Engine/Server/Src/GameEngine/Application/Application.cpp

```diff
@@ -139,8 +142,23 @@ GameEngine::Application::Application()
     }, *_registries);
 }
 
+bool GameEngine::Application::noPlayerConnected()
+{
+    ECS::SparseArray<IComponent> PlayerComponentArray = _registries->get_components<IComponent>("PlayerComponent");
+    for (std::size_t index = 0; index < PlayerComponentArray.size(); index++) {
+        PlayerComponent* player = dynamic_cast<PlayerComponent*>(PlayerComponentArray[index].get());
+        if (player && player->token != 0) {
+            return false;
+        }
+    }
+    return true;
+}
+
 void GameEngine::Application::run()
 {
+    while (noPlayerConnected())
+        SLEEP(0.5);
+
     while (true)
```

Comment:
We keep this While true ?

Reply:
It's to start the server only when a first player is connected.

Reply:
But I will change something.